test/jdk/javax/net/ssl/ExtendedSSLSession/ExportKeyingMaterialTests.java

```diff
@@ -0,0 +1,586 @@
+
+            // After we've transferred all application data between the client
+            // and server, we close the clientEngine's outbound stream.
+            // This generates a close_notify handshake message, which the
+            // server engine receives and responds by closing itself.
+            if (!dataDone && (clientOut.limit() == serverIn.position()) &&
+                    (serverOut.limit() == clientIn.position())) {
+
+                runExporterTests(
+                        (ExtendedSSLSession) clientEngine.getSession(),
+                        (ExtendedSSLSession) serverEngine.getSession());
+
+                // A sanity check to ensure we got what was sent.
+                checkTransfer(serverOut, clientIn);
+                checkTransfer(clientOut, serverIn);
+
+                log("\tClosing clientEngine's *OUTBOUND*...");
+                clientEngine.closeOutbound();
+                logEngineStatus(clientEngine);
+
+                dataDone = true;
+                log("\tClosing serverEngine's *OUTBOUND*...");
+                serverEngine.closeOutbound();
+                logEngineStatus(serverEngine);
+            }
+        }
+    }
+
+    private static void runExporterTests(
+            ExtendedSSLSession clientSession,
+            ExtendedSSLSession serverSession) throws Exception {
+
+        SecretKey clientKey, serverKey;
+
+        // Create output arrays
+        byte[] clientBytes, serverBytes;
+
+        // Create various input arrays and fill with junk.
+        Random random = new Random();
+        byte[] bytes = new byte[20];
+        random.nextBytes(bytes);
+
+        // Slightly change 1 byte in the middle
+        byte[] bytesDiff = Arrays.copyOf(bytes, bytes.length);
+        bytesDiff[bytes.length/2]++;
+
+        byte[] bytesDiffSize = new byte[21];
+        random.nextBytes(bytesDiffSize);
+
+        // Run a bunch of similar derivations using both the Key/Data methods,
+        // exercising the various valid/invalid combinations.
+
+        // We may need to adjust if it turns out that this is run with
+        // non-extractable keys if .equals() doesn't work.
+
+        // Inputs exactly equal.
+        clientKey = clientSession.exportKeyingMaterialKey("hello", bytes, 128);
+        serverKey = serverSession.exportKeyingMaterialKey("hello", bytes, 128);
+        assertEquals(clientKey, serverKey,
+                "Key: Equal inputs but exporters are not equal");
+        log("Key: Equal inputs test passed");
+
+        clientBytes = clientSession.exportKeyingMaterialData("hello",
+                bytes, 128);
+        serverBytes = serverSession.exportKeyingMaterialData("hello",
+                bytes, 128);
+        assertEqualsByteArray(clientBytes, serverBytes,
+                "Data: Equal inputs but exporters are not equal");
+        log("Data: Equal inputs test passed");
+
+        // Different labels, now use exportKeyMaterialData() for coverage
```

Comment:
Should update the comment with new method name.

Reply:
Done